// @storybook/addon-info/dist/components/markdown/pre/copyButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var Button =
/*#__PURE__*/
(0, _styled.default)("button", {
  target: "eva467m0"
})({
  overflow: 'hidden',
  border: '1px solid #eee',
  borderRadius: 3,
  backgroundColor: '#FFFFFF',
  cursor: 'pointer',
  fontSize: 13,
  padding: '3px 10px',
  alignSelf: 'flex-start',
  flexShrink: 0,
  ':hover': {
    backgroundColor: '#f4f7fa',
    borderColor: '#ddd'
  },
  ':active': {
    backgroundColor: '#e9ecef',
    borderColor: '#ccc'
  }
}, function (_ref) {
  var styles = _ref.styles;
  return styles;
});
var ContentWrapper =
/*#__PURE__*/
(0, _styled.default)("div", {
  target: "eva467m1"
})({
  transition: 'transform .2s ease',
  height: 16
}, function (_ref2) {
  var styles = _ref2.styles,
      toggled = _ref2.toggled;
  return (0, _objectSpread2.default)({}, styles, {
    transform: toggled ? 'translateY(0px)' : 'translateY(-100%) translateY(-6px)'
  });
});

function CopyButton(_ref3) {
  var theme = _ref3.theme,
      onClick = _ref3.onClick,
      toggled = _ref3.toggled;
  var _theme$copyButton = theme.copyButton,
      copyButton = _theme$copyButton === void 0 ? {} : _theme$copyButton,
      copyButtonContent = theme.copyButtonContent;
  var _copyButton$toggleTex = copyButton.toggleText,
      toggleText = _copyButton$toggleTex === void 0 ? 'Copied!' : _copyButton$toggleTex,
      _copyButton$text = copyButton.text,
      text = _copyButton$text === void 0 ? 'Copy' : _copyButton$text,
      copyButtonStyles = (0, _objectWithoutProperties2.default)(copyButton, ["toggleText", "text"]);
  return _react.default.createElement(Button, {
    onClick: onClick,
    styles: copyButtonStyles
  }, _react.default.createElement(ContentWrapper, {
    styles: copyButtonContent,
    toggled: toggled
  }, _react.default.createElement("div", {
    style: {
      marginBottom: 6
    }
  }, toggleText), _react.default.createElement("div", null, text)));
}

CopyButton.propTypes = {
  onClick: _propTypes.default.func,
  toggled: _propTypes.default.bool,
  theme: _propTypes.default.shape({
    copyButton: _propTypes.default.object
  })
};
CopyButton.defaultProps = {
  onClick: function onClick() {},
  toggled: false,
  theme: {}
};
var _default = CopyButton;
exports.default = _default;
