// @storybook/addon-info/dist/components/markdown/pre/pre.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styled = _interopRequireDefault(require("@emotion/styled"));

var _copyButton = _interopRequireDefault(require("./copyButton"));

var _copy = _interopRequireDefault(require("./copy"));

var TOGGLE_TIMEOUT = 1800;
var StyledPre =
/*#__PURE__*/
(0, _styled.default)("pre", {
  target: "eyvlbql0"
})({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  fontSize: '.88em',
  fontFamily: 'Menlo, Monaco, "Courier New", monospace',
  backgroundColor: '#fafafa',
  padding: '.5rem',
  lineHeight: 1.5,
  overflowX: 'scroll'
}, function (_ref) {
  var styles = _ref.styles;
  return styles;
});

var Pre =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Pre, _React$Component);

  function Pre() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Pre);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Pre)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {
      copied: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "setRef", function (elem) {
      _this.pre = elem;
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleClick", function () {
      var text = _this.pre && _this.pre.innerText;

      if (!text) {
        return;
      }

      (0, _copy.default)(text);

      _this.setState({
        copied: true
      });

      clearTimeout(_this.timeout);
      _this.timeout = setTimeout(function () {
        _this.setState({
          copied: false
        });
      }, TOGGLE_TIMEOUT);
    });
    return _this;
  }

  (0, _createClass2.default)(Pre, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          theme = _this$props.theme,
          children = _this$props.children;
      var pre = theme.pre;
      var copied = this.state.copied;
      return _react.default.createElement(StyledPre, {
        styles: pre
      }, _react.default.createElement("div", {
        ref: this.setRef
      }, children), _react.default.createElement(_copyButton.default, {
        onClick: this.handleClick,
        toggled: copied
      }));
    }
  }]);
  return Pre;
}(_react.default.Component);

Pre.propTypes = {
  children: _propTypes.default.node,
  theme: _propTypes.default.shape({
    pre: _propTypes.default.object
  })
};
Pre.defaultProps = {
  children: null,
  theme: {}
};
var _default = Pre;
exports.default = _default;
