// @storybook/addon-info/dist/components/markdown/code.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Blockquote = Blockquote;
Object.defineProperty(exports, "Pre", {
  enumerable: true,
  get: function get() {
    return _pre.default;
  }
});
exports.Code = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _global = require("global");

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _pre = _interopRequireDefault(require("./pre/pre"));

var Code =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Code, _React$Component);

  function Code() {
    (0, _classCallCheck2.default)(this, Code);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Code).apply(this, arguments));
  }

  (0, _createClass2.default)(Code, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.highlight();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      this.highlight();
    }
  }, {
    key: "highlight",
    value: function highlight() {
      if (typeof _global.Prism !== 'undefined') {
        _global.Prism.highlightAll();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          language = _this$props.language,
          code = _this$props.code;
      var codeStyle = {
        fontFamily: 'Menlo, Monaco, "Courier New", monospace',
        backgroundColor: '#fafafa'
      };
      var preStyle = {
        fontFamily: 'Menlo, Monaco, "Courier New", monospace',
        backgroundColor: '#fafafa',
        padding: '.5rem',
        lineHeight: 1.5,
        overflowX: 'scroll'
      };
      var className = language ? "language-".concat(language) : '';
      return _react.default.createElement("pre", {
        style: preStyle,
        className: className
      }, _react.default.createElement("code", {
        style: codeStyle,
        className: className
      }, code));
    }
  }]);
  return Code;
}(_react.default.Component);

exports.Code = Code;
Code.propTypes = {
  language: _propTypes.default.string,
  code: _propTypes.default.node
};
Code.defaultProps = {
  language: null,
  code: null
};

function Blockquote(_ref) {
  var children = _ref.children;
  var style = {
    fontSize: '1.88em',
    fontFamily: 'Menlo, Monaco, "Courier New", monospace',
    borderLeft: '8px solid #fafafa',
    padding: '1rem'
  };
  return _react.default.createElement("blockquote", {
    style: style
  }, children);
}

Blockquote.propTypes = {
  children: _propTypes.default.node
};
Blockquote.defaultProps = {
  children: null
};
